builder/vsphere/common/step_reattach_cdrom_test.go

```diff
@@ -43,23 +43,68 @@ func TestStepReattachCDRom_Run(t *testing.T) {
 			},
 			expectedVmMock: &driver.VirtualMachineMock{
 				EjectCdromsCalled:        true,
-				RemoveCdromsCalled:       true,
 				ReattachCDRomsCalled:     true,
 				FindSATAControllerCalled: true,
-				AddCdromCalledTimes:      8,
+				AddCdromCalledTimes:      6,
 				AddCdromTypes: []string{
 					"sata", "sata",
 					"sata", "sata",
 					"sata", "sata",
-					"sata", "sata",
 				},
-				AddCdromPaths: []string{
-					"[datastore] /iso/linux.iso", "[datastore] /iso/linux.iso",
-					"[datastore] /iso/linux.iso", "[datastore] /iso/linux.iso",
-					"[datastore] /iso/linux.iso", "[datastore] /iso/linux.iso",
-					"[datastore] /iso/linux.iso", "[datastore] /iso/linux.iso",
+			},
+			fail: false,
+		},
+		{
+			name: "Reattach 2 less CD-ROM devices than initially added",
+			step: &StepReattachCDRom{
+				Config: &ReattachCDRomConfig{
+					ReattachCDRom: 2,
+				},
+				CDRomConfig: &CDRomConfig{
+					CdromType: "sata",
+					ISOPaths: []string{
+						"[datastore] /iso/linux1.iso",
+						"[datastore] /iso/linux2.iso",
+						"[datastore] /iso/linux3.iso",
+						"[datastore] /iso/linux4.iso",
+					},
+				},
+			},
+			expectedAction: multistep.ActionContinue,
+			vmMock: &driver.VirtualMachineMock{
+				ReattachCDRomsCalled: true,
+			},
+			expectedVmMock: &driver.VirtualMachineMock{
+				RemoveNCdromsCalled:  true,
+				EjectCdromsCalled:    true,
+				ReattachCDRomsCalled: true,
+				AddCdromCalledTimes:  0,
+			},
+			fail: false,
+		},
+		{
+			name: "Reattach same amount of CD-ROMs as initially added",
```

Comment:
```suggestion
			name: "Successfully attach the same number of CD-ROMs as initially added.",
```

Reply:
Done! I removed a dot at the end because from looking at other tests I see the sentence ends with no dot at the end. Please tell if you want me to add it.

Reply:
Resolved in 860f7101405d68474e1acadf652ef640f3e43b2b.